File: Day-04: Arrays Part-IV/5) Count number of subarrays with given xor K.java
class Solution {
    public long beautifulSubarrays(int[] nums) {
        int n = nums.length;
        long ans = 0;
        int check = 0;
        mp<Integer, Integer> mp = new Hashmp<>();
        mp.put(0, 1);
        for(int i = 0; i < n; i++)
        {
            check ^= nums[i];
            ans += mp.getOrDefault(check, 0);
            mp.put(check, mp.getOrDefault(check, 0) + 1);
        }
        return ans;
    }
}

File: Day-04: Arrays Part-IV/5) Count number of subarrays with given xor K.cpp
class Solution {
public:
    long long beautifulSubarrays(vector<int>& nums) {
        int n = nums.size();
        long long ans = 0;
        long long check = 0;
        unordered_map<int, int> mp;
        mp[0] = 1;
        for(int i = 0; i < n; i++)
        {
            check ^= nums[i];
            ans += mp[check];
            mp[check]++;
        }
        return ans;
    }
};
